typos.toml

```diff
@@ -0,0 +1,41 @@
+[files]
+extend-exclude = [
+    # ./compiler excludes
+    "compiler/rustc_codegen_gcc",
+    "compiler/rustc_codegen_cranelift",
+    "compiler/rustc_baked_icu_data",
+]
+
+[default.extend-words]
```

Comment:
What is the meaning of this section? All of the entries here are `x = "x"`, so this doesn't seem to *extend* anything? Would be worth a comment so the file can be more easily edited by people that are not familiar with the underlying tooling.

Reply:
Added description and link to docs.